README.md

```diff
@@ -6,16 +6,133 @@
+
+### Getting the contract documentation
+
+Use `homebase-lite document`. That will produce `homebase-lite.md` file in the current working directory. This documentation is also attached to binary releases.
+
+You can use `--output FILEPATH` key to specify a different output path, or use `-` to print to standard output.
+
+### Deploying the contract
+
+You could print the contract's Michelson code and initial storage and then just use `tezos-client originate` if you wanted to, but `homebase-lite` executable provides a more convenient option: `homebase-lite originate`. You will need to have `tezos-client` binary available on your system.
+
+You'll need to specify at least FA2 contract address and administrator's keyhash. You also might need to specify Tezos node endpoint URL with `--endpoint` (but if you don't, the one from `tezos-client` config will be used):
+
+```bash
+$ homebase-lite originate --fa2-address KT1AbgvM5D1wAVZbsB3WTCB54E94p2hn1Rb9 --admin tz1Znvao83anTh654Uu44Kb9atTp5yTK93A8
+```
+
+If tezos-client binary is not in PATH, you will need to specify the full path to the executable with `--client-path` option, e.g.
+
+```bash
+$ homebase-lite originate --fa2-address KT1AbgvM5D1wAVZbsB3WTCB54E94p2hn1Rb9 --admin tz1Znvao83anTh654Uu44Kb9atTp5yTK93A8 --client-path /path/to/tezos-client
+```
+
+Note that the origination is performed as the admin address, so you can not specify the admin address for an account you don't have the private key for. Also, obviously, the admin account needs enough tz for the origination.
+
+Please also note there is no way to change the fa2 contract address after the contract is originated, so double-check if it's correct.
+
+Upon successful origination, the contract's address is printed to stdout and stored in `tezos-client`'s alias store with the alias `homebase-lite`. Note that if the alias already exists it will be overwritten. You can specify a different alias with the `--name` key.
+
+You can also specify FA2 token id and any other initial storage parameters. See `homebase-lite originate --help` for more information.
+
+## Running the tests
+
+If you want to run the included automated testsuite, you'll need to first [build from source](#build-instructions).
+
+There are two kind of tests: emulated tests and network tests.
+
+To run emulated tests only, in the root of the repository run `stack test`. This will build the tests if needed and run them, printing the report to standard output.
+
+To run the network tests, you will need to setup an implicit account with plenty of tz to run tests, as repeated originations can get quite costly (so you probably want to run those on the testnet). You will also need `tezos-client` binary.
+
+To run network tests only, use the following command:
+
+```bash
+$ stack test --ta '--cleveland-mode only-network --cleveland-node-endpoint https://testnet.endpoint.url/ --cleveland-data-dir /path/to/temporary/directory --cleveland-moneybag-secret-key unencrypted:edsk...'
+```
```

Comment:
How about adding a `Makefile` target to make calling this easier?

Reply:
The issue is, apart from `--cleveland-mode only-network`, the user has to specify the rest anyway. Well, maybe except the tempdir, which could be created in the makefile (but honestly that's a bit annoying in itself, due to the idiosyncrasies of Make and the zoo of various `mktemp`-like commands out there, way easier to do this with a shell script). The only thing we'd be sidestepping here is argument escaping.

Reply:
Yeahhh... that's fair.

Ok, let's not do that, but can you modify it to a multiline command?
I think it would at least look a little bit less scary that way.

Reply:
Yeah... that's a little bit tricky, too, because of how `stack` handles `--test-arguments`, one spurious space somewhere and stack will complain about unknown options, which would be rather confusing. Okay, I'll try to figure something out with the makefile.

Reply:
Okay, I did _something_ that looks arguably less scary.